src/networking/systems.rs

```diff
@@ -0,0 +1,240 @@
+            ReceiveEvent::SocketClosed => {
+
+                // Delete all connection entities
+                for (_, e) in entities.connections.drain() {
+                    commands.despawn(e);
+                }
+
+                // drop all stream senders
+                session.stream_senders.clear();
+            }
+
+            _ => {}
+        }
+
+        // Publish event for other systems to consume
+        net_events.send(event);
+    }
+}
+
+// Take ECS events and forward them to MPSCs to be sent over the network
+pub fn send_net_events(mut session: ResMut<SessionEventListenerState>, send_events: ResMut<Events<SendEvent>>) {
+    // Publish packets ready to send to appropriate MPSC channels
+    for send in session.send_event_reader.iter(&send_events) {
+        match send {
+            SendEvent::SendPacket { connection_id, stream_id, data } => {
+                let sender = session
+                    .stream_senders
+                    .entry(connection_id.clone())
+                    .or_default()
+                    .get(&stream_id);
+
+                if let Some(sender) = sender {
+                    if let Err(e) = sender.send(data.to_owned()) {
+                        error!("Send Error: {:?}", e);
+                    }
+                } else {
+                    error!(
+                        "Attempted to send to a non-existant stream: {:?}:{:?}",
+                        connection_id, stream_id
+                    );
+                }
+            }
+        }
+    }
+}
+
+// Handle all the work of a single connection (waiting for new streams to open)
+pub async fn handle_connection(
+    conn: quinn::Connecting,
+    event_sender: UnboundedSender<ReceiveEvent>,
+) -> Result<(), ()> {
+    // Generate a unique ID for this connection
+    let guid = ConnectionId::new();
+    info!("connection incoming: {:?}", guid);
+
+    // Wait for connection to finish connecting
+    let quinn::NewConnection { mut bi_streams, .. } = conn
+        .await
+        .map_err(|e| error!("Connection Error: {:?}", e))?;
+
+    // Send an intial event indicating that this connection opened
+    event_sender
+        .send(ReceiveEvent::Connected(guid))
+        .expect("Failed to send network event");
+
+    // Keep getting events from the connection until it closes
+    while let Some(stream) = bi_streams.next().await {
+        match stream {
+            Err(quinn::ConnectionError::ApplicationClosed { .. }) => break,
+            Err(e) => error!("Connection error: {:?}", e),
+            Ok((send, recv)) => {
+                tokio::spawn(handle_stream(guid, send, recv, event_sender.clone()));
+            }
+        };
+    }
+
+    // Send a final event indicating that this connection closed
+    event_sender
+        .send(ReceiveEvent::Disconnected(guid))
+        .expect("Failed to send network event");
+
+    Ok(())
+}
+
+// Handle all the work of a specific stream
```

Comment:
```suggestion
/// Handle all the work of a specific stream
```
Might as well turn this into a doc comment since you already did the work to explain it